test/e2e/util/util.go

```diff
@@ -757,6 +757,35 @@ func EnsureNetworkPolicies(t *testing.T, ctx context.Context, c crclient.Client,
 	})
 }
 
+// EnsureNodesRuntime ensures that all nodes in the NodePool have the expected runtime handlers.
+// This is only supported on 4.18+ when the default runtime is changed to crun.
+func EnsureNodesRuntime(t *testing.T, ctx context.Context, c crclient.Client, nodepool *hyperv1.NodePool) {
+	AtLeast(t, Version418)
+	g := NewWithT(t)
+
+	nodeList := &corev1.NodeList{}
+	err := c.List(ctx, nodeList)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to list nodes")
+
+	validHandlers := map[string]bool{
+		"runc": false,
+		"crun": false,
+	}
+	for _, node := range nodeList.Items {
+		if node.Status.RuntimeHandlers != nil {
+			for _, handler := range node.Status.RuntimeHandlers {
```

Comment:
fwiw you could something like

```
	for _, node := range nodeList.Items {
		if node.Status.RuntimeHandlers != nil {
			fail
		}
		Expect(node.Status.RuntimeHandlers).To(ConsistOf(
			NodeRuntimeHandler{Name: "runc"},
			NodeRuntimeHandler{Name: "crun"},
		))
	}
```

Reply:
I like it, but we need to upgrade the OAPI dependency in order to do this kind of comparison. I can do that if you agree. There is another field called UserNamespaces in the [NodeRuntimeHandlerFeatures](https://pkg.go.dev/k8s.io/api@v0.31.1/core/v1#NodeRuntimeHandlerFeatures) struct which needs to be set for that comparision. WDYT?